src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -1011,11 +1034,73 @@ emit_arm64_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatur
 				op = SIMD_OP_LLVM_I64ABS;
 				break;
 			}
+		}
 
-			return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X, op, arg0_type, fsig, args);
+		case SN_AbsoluteCompareGreaterThan: {
+			return emit_absolute_compare(cfg, klass, fsig, arg0_type, args, SIMD_OP_LLVM_I4FABSOLUTE_COMPARE_GREATER_THAN, SIMD_OP_LLVM_DABSOLUTE_COMPARE_GREATER_THAN);
 		}
+
+	    case SN_AbsoluteCompareGreaterThanOrEqual: {
+			return emit_absolute_compare(cfg, klass, fsig, arg0_type, args, SIMD_OP_LLVM_FABSOLUTE_COMPARE_GREATER_THAN_OR_EQUAL, SIMD_OP_LLVM_DABSOLUTE_COMPARE_GREATER_THAN_OR_EQUAL);
+		}
+
+		case SN_AbsoluteCompareLessThan: {
+			//Compare less than uses the same instructions as greater than, with arguments swapped.
+			MonoInst* temp_for_swap = args[0];
+			args[0] = args[1];
+			args[1] = temp_for_swap;
+
+			return emit_absolute_compare(cfg, klass, fsig, arg0_type, args, SIMD_OP_LLVM_FABSOLUTE_COMPARE_LESS_THAN, SIMD_OP_LLVM_DABSOLUTE_COMPARE_LESS_THAN);
+		}
+
+		case SN_AbsoluteCompareLessThanOrEqual: {
+			//Compare less than uses the same instructions as greater than, with arguments swapped.
+			MonoInst* temp_for_swap = args[0];
+			args[0] = args[1];
+			args[1] = temp_for_swap;
+
+			return emit_absolute_compare(cfg, klass, fsig, arg0_type, args, SIMD_OP_LLVM_FABSOLUTE_COMPARE_LESS_THAN_OR_EQUAL, SIMD_OP_LLVM_DABSOLUTE_COMPARE_LESS_THAN_OR_EQUAL);
 		}
 		
+		case SN_AbsSaturate:{
+			SimdOp op = (SimdOp)0;
+			switch(get_underlying_type(fsig->params [0])) {
+			case MONO_TYPE_I1:
+				op = SIMD_OP_LLVM_I8ABS_SATURATE;
+				break;
+			case MONO_TYPE_I2:
+				op = SIMD_OP_LLVM_I16ABS_SATURATE;
+				break;
+			case MONO_TYPE_I4:
+				op = SIMD_OP_LLVM_I32ABS_SATURATE;
+				break;
+			case MONO_TYPE_I8:
+				op = SIMD_OP_LLVM_I64ABS_SATURATE;
+				break;
+			}
+
+			return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X, op, arg0_type, fsig, args);
+		}
+
+		case SN_AbsScalar: {
+			SimdOp op = (SimdOp)0;
+			switch(get_underlying_type(fsig->params [0])) {
+			case MONO_TYPE_I1:
+				op = SIMD_OP_LLVM_I8ABS_SATURATE;
+				break;
+			case MONO_TYPE_I2:
+				op = SIMD_OP_LLVM_I16ABS_SATURATE;
+				break;
+			case MONO_TYPE_I4:
+				op = SIMD_OP_LLVM_I32ABS_SATURATE;
+				break;
+			case MONO_TYPE_I8:
+				op = SIMD_OP_LLVM_I64ABS_SATURATE;
+				break;
+			}
+			return emit_simd_ins_for_sig (cfg, klass, OP_XOP_X_X, op, arg0_type, fsig, args);
+		}		
+		}
```

Comment:
Why is this extra curly brace needed? Am I just blind?

Reply:
The first curly brace and the case SN_AbsScalar: { block, and the other one ends the outer switch statement

I don't like it either, but if we want to go with the style of cases at the same level as a switch, and have curly braces around the larger cases, it seems like this is the logical result.